src/text/text_formatter.rs

```diff
@@ -278,24 +279,29 @@ impl<'a, W: std::fmt::Write> FmtValueFormatter<'a, W> {
     }
 
     pub(crate) fn format_symbol_token<A: AsRawSymbolRef>(&mut self, token: A) -> IonResult<()> {
-        let text = match token.as_raw_symbol_ref() {
-            RawSymbolRef::SymbolId(sid) => return Ok(write!(self.output, "${sid}")?),
-            RawSymbolRef::SystemSymbol_1_1(system_symbol) => system_symbol.text(),
-            RawSymbolRef::Text(text) => text,
+        use RawSymbolRef::*;
+        let write_result = match token.as_raw_symbol_ref() {
+            SymbolId(sid) => write!(self.output, "${sid}"),
+            // '' is the only system symbol that requires quoting; the rest are identifiers.
+            SystemSymbol_1_1(v1_1::system_symbols::EMPTY_TEXT) => write!(self.output, "\'\'"),
+            // Any other system symbol is an identifier and doesn't require quoting.
+            SystemSymbol_1_1(symbol) => return Ok(write!(self.output, "{}", symbol.text())?),
```

Comment:
Now that it's been refactored to have the `match` expression return a write result, you could get rid of the early return here, right?
```suggestion
            SystemSymbol_1_1(symbol) => write!(self.output, "{}", symbol.text()),
```